README.md

```diff
@@ -26,10 +26,57 @@ gem install faraday-retry-failover
 
 ## Usage
 
+This extension can be used on it's own but is ultimately meant to be used in tandem with the [Faraday::Retry](https://github.com/lostisland/faraday-retry) middleware.
+You should make sure that this middleware is added in the request stack ***after*** the retry middleware to ensure
+it can alter the environment on retries.
+
+When you configure the connection, you can do it a few different ways for the failover to work.
+
+### Setting the URL in the request
+
+This simplest way is to make a generic Faraday connection, and pass the full URI in the request.
+When this is done, the failover middleware will parse the URI and use the host and port to determine
+the failover endpoints using DNS resolution.
+
 ```ruby
 require 'faraday/retry/failover'
 
-# TODO
+conn = Faraday.new do |f|
+  f.request :retry, { max: 2 }
+  f.request :retry_with_failover
+end
+
+conn.get('https://api.invoca.net/api/2020-10-01/transaction/33.json')
+```
+
+### Setting the base URL in the connection
+
+This is the same as the previous example, but the base URL is set in the connection. This is useful
+when the base URL is always the same, and you have a DNS entry that will resolve to multiple IPs.
+
+```ruby
+require 'faraday/retry/failover'
+
+conn = Faraday.new('https://api.invoca.net') do |f|
+  f.request :retry, { max: 2 }
+  f.request :retry_with_failover
```

Comment:
As discussed offline, I've gone ahead and renamed the gem to `Faraday::HighlyAvailableRetries` to be more descriptive of what it's actually doing. Documentation has also been updated to match.

To answer your other question, when you have 3 retries but only 2 resolved endpoints like you described, it will do `[A, B, A]` rather than short-circuiting after exhausting the endpoint list. That explanation has been added to the README.

Reply:
Excellent. I think the `[A, B, A]` approach is the least surprising.